src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -671,7 +671,7 @@ public URL getUrl() throws MalformedURLException {
         if (port == 0) {
             throw new IllegalStateException("This method must be called after calling #startJenkins.");
         }
-        return new URL("http://" + host + ":" + port + "/jenkins/");
```

Comment:
This failed when `host` was an IPV6 ip

Reply:
I think this line was the only necessary change. The other change caused test regressions in other parts, so I'm proposing a revert of them in https://github.com/jenkinsci/jenkins-test-harness/pull/776